draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -6407,18 +6407,29 @@ Expect: 100-continue
 </ol>
 <t>
    An origin server &MUST-NOT; perform the requested method if a received
-   If-Match condition evaluates to false; instead, the origin server &MUST;
-   respond with either
-   a) the <x:ref>412 (Precondition Failed)</x:ref> status code or
-   b) one of the <x:ref>2xx (Successful)</x:ref> status codes if the origin
-   server has verified that a state change is being requested and the final
-   state is already reflected in the current state of the target resource
+   If-Match condition evaluates to false. Instead, the origin server &MAY;
+   indicate that the conditional request failed by responding with a
+   <x:ref>412 (Precondition Failed)</x:ref> status code. Alternatively,
+   if the request is a state-changing operation that appears to have already
+   been applied to the selected representation, the origin server &MAY; respond
+   with a <x:ref>2xx (Successful)</x:ref> status code
```

Comment:
Here, we are replacing a "MUST reply with 412 or 2xx" with "MAY reply with 412 or MAY reply with 2xx". Not sure whether this change of wording is good.

Reply:
The original wording is wrong because a server can send any code to the client, for any reason, and it only effects interoperability if the code doesn't reflect the server's action. For example, timeouts, invalid credentials, server error, etc. What an origin server must do is a lot more complicated that an either/or phrase.

Reply:
An alternative would be to write these as "if the server responds with 2xx, it means ..." kind of statements.